pymc3/distributions/discrete.py

```diff
@@ -391,19 +424,15 @@ def logp(value, p):
         -------
         TensorVariable
         """
-        # if self._is_logit:
-        #     lp = at.switch(value, self._logit_p, -self._logit_p)
-        #     return -log1pexp(-lp)
-        # else:
+        lp = at.switch(value, -logit_p, logit_p)
         return bound(
-            at.switch(value, at.log(p), at.log(1 - p)),
-            value >= 0,
+            -log1pexp(lp),
+            0 <= value,
             value <= 1,
-            p >= 0,
-            p <= 1,
+            ~at.isnan(logit_p),
```

Comment:
Invalid `p` gets converted to `nan` so this is how we can identify it in `bound`